### PR TITLE
Migrate samples build to GitHub actions

### PR DESCRIPTION
**Pull request by GMNGeoffrey**
This is a direct port of the Buildkite build, without much attempt at
optimizing or doing some things more intelligently (e.g. reusing a
previous host build, using this for release validation, etc). Future
work may want to reshuffle things here, but this removes our last usage
of Buildkite that isn't benchmarks.

Part of https://github.com/iree-org/iree/issues/9855

skip-ci: This does not affect the main CI build

### CODE REVIEW

File: .github/workflows/samples.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+# Workflow for running Examples of IREE usage against releases periodically.
+
+name: samples
+
+on:
+  schedule:
+    - cron: "0 13 * * *"
+  workflow_dispatch:
+  # DO NOT SUBMIT: Turn off presubmit build
+  pull_request:
```
Comment (GMNGeoffrey) [inline .github/workflows/samples.yml:16]: Reminder to revert

File: .github/workflows/samples.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        uses: actions/checkout@e2f20e631ae6d7dd3b768f56a5d2af784dd54791 # v2.5.0
+      - name: "Testing Samples"
+        run: |
+          ./build_tools/github_actions/docker_run.sh  \
+            gcr.io/iree-oss/samples@sha256:291d87d125b41e1539dbb96c7c4f734cc98fc158a36d18496cda360658f00bde \
+            ./build_tools/testing/test_samples.sh
+
+  # FIXME: unify this with host build from ci.yml. This is complicated by
+  # incompatibility between storage options for GitHubmanaged and self-hosted
+  # runners.
```
Comment (ScottTodd) [inline .github/workflows/samples.yml:43]: nit: missing a space or dash between these words

```suggestion
  # FIXME: unify this with host build from ci.yml. This is complicated by
  # incompatibility between storage options for GitHub-managed and self-hosted
  # runners.
```

File: .github/workflows/samples.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  wasm:
```
Comment (ScottTodd) [inline .github/workflows/samples.yml:44]: This will include WebGPU and WebAssembly code, so maybe rename to just "web"
```suggestion
  web:
```

File: .github/workflows/samples.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# Workflow for running Examples of IREE usage against releases periodically.
+
+name: samples
+
+on:
+  schedule:
+    - cron: "0 13 * * *"
```
Comment (ScottTodd) [inline .github/workflows/samples.yml:13]: What's this in plain English? (PST)
We had the Buildkite pipeline running around 4-5pm PST every day, but we talked about running earlier in the day (and maybe only on weekdays)

**Review by ScottTodd — APPROVED** [review]

File: .github/workflows/samples.yml (hunk 4 above)
Comment (GMNGeoffrey) [reply .github/workflows/samples.yml:13]: It's early morning PT. Do we want it later? (Also added a comment with the time)

Comment (ScottTodd) [reply .github/workflows/samples.yml:13]: Early morning PT sounds fine to me. I didn't really like getting emails at 5pm on a Friday about build failures, since they'd be nagging at me over the weekend.

File: experimental/web/sample_dynamic/build_sample.sh [hunk 5]
```diff
@@ -14,8 +14,29 @@
...
+set -euo pipefail
+
+ROOT_DIR=$(git rev-parse --show-toplevel)
+
+HOST_BUILD_DIR="${IREE_HOST_BUILD_DIR:-${ROOT_DIR}/build-host}"
+BUILD_DIR="${IREE_EMPSCRIPITEN_BUILD_DIR:-build-emscripten}"
+INSTALL_ROOT="${1:-${HOST_BUILD_DIR}/install}"
+CMAKE_BIN=${CMAKE_BIN:-$(which cmake)}
+SOURCE_DIR=${ROOT_DIR}/experimental/web/sample_dynamic
+BINARY_DIR=${BUILD_DIR}/experimental/web/sample_dynamic
```
Comment (pzread) [inline experimental/web/sample_dynamic/build_sample.sh:38]: Any chance to use `source build_tools/cmake/setup_build.sh` here?

File: experimental/web/sample_static/build_sample.sh [hunk 6]
```diff
@@ -27,42 +48,39 @@ then
...
-CMAKE_BIN=${CMAKE_BIN:-$(which cmake)}
-ROOT_DIR=$(git rev-parse --show-toplevel)
-SOURCE_DIR=${ROOT_DIR}/experimental/web/sample_static
-
-BUILD_DIR=${ROOT_DIR?}/build-emscripten
-mkdir -p ${BUILD_DIR}
-
-BINARY_DIR=${BUILD_DIR}/experimental/web/sample_static/
-mkdir -p ${BINARY_DIR}
+if [[ -d "${BUILD_DIR}" ]]; then
```
Comment (pzread) [inline experimental/web/sample_static/build_sample.sh:51]: Same here.

**Review by pzread — APPROVED** [review]

File: experimental/web/sample_dynamic/build_sample.sh [hunk 7]
```diff
@@ -14,8 +14,29 @@
 #
 # Usage:
 #   build_sample.sh (optional install path) && serve_sample.sh
+#
+# The desired host install directory can be passed as the first argument.
+# Otherwise, it looks for an install directory under path set in the environment
+# variable IREE_HOST_BUILD_DIR (default build-host). The build directory for the
+# empscripten build is taken from the environment variable
+# IREE_EMPSCRIPITEN_BUILD_DIR, defaulting to "build-empscripten". Designed for
```
Comment (ScottTodd) [inline experimental/web/sample_dynamic/build_sample.sh:22]: Typos, these should be "emscripten", not "empscripten" or "EMPSCRIPITEN"

https://emscripten.org/

File: experimental/web/sample_dynamic/build_sample.sh (hunk 5 above)
Comment (ScottTodd) [reply experimental/web/sample_dynamic/build_sample.sh:38]: Possibly... though these scripts interact with the build directory more closely than other scripts do

File: experimental/web/sample_dynamic/build_sample.sh (hunk 7 above)
Comment (GMNGeoffrey) [reply experimental/web/sample_dynamic/build_sample.sh]: Wow pretty impressive that I managed to misspell this in two different ways.

File: experimental/web/sample_dynamic/build_sample.sh (hunk 5 above)
Comment (GMNGeoffrey) [reply experimental/web/sample_dynamic/build_sample.sh]: I looked at using that, but ended up not because this script has other requirements about being runnable manually, which mostly consrains assuming we're in the root directory to start.

Comment (GMNGeoffrey) [reply experimental/web/sample_dynamic/build_sample.sh]: I looked at using that and then decided not to because this script has other requirements about being runnable manually, which mostly means not assuming we're in the root directory to start. But looking again, it should be fine, since that script doesn't do anything related to the root directory or other paths.

**Review by ScottTodd — APPROVED** [review]
LGTM (just remember to revert the `pull_request` trigger)
